internal/ignition/ignition.go

```diff
@@ -1359,7 +1359,8 @@ func (ib *ignitionBuilder) FormatDiscoveryIgnitionFile(ctx context.Context, infr
 		ignitionParams["ServiceIPs"] = dataurl.EncodeBytes([]byte(GetServiceIPHostnames(cfg.ServiceIPs)))
 	}
 
-	if infraEnv.StaticNetworkConfig != "" && common.ImageTypeValue(infraEnv.Type) == models.ImageTypeFullIso {
+	shouldAddStaticNetworkConfig := string(overrideDiscoveryISOType) == string(DiscoveryISOTypeFullIso) || (string(overrideDiscoveryISOType) == string(DiscoveryISOTypeUseInfraEnv) && infraEnv.StaticNetworkConfig != "" && common.ImageTypeValue(infraEnv.Type) == models.ImageTypeFullIso)
+	if shouldAddStaticNetworkConfig {
```

Comment:
I think this would be more clear as multiple steps.

```
isoType := overrideDiscoveryISOType
if isoType == "" {
  isoType = infraEnv.Type
}
if infraEnv.StaticNetworkConfig != "" && common.ImageTypeValue(isoType) == models.ImageTypeFullIso {
...
```

This more clearly shows that we're prioritizing the function parameter over the infra env and separates out the check for actually having static networking.

It's less efficient, but I think the readability is worth it in this case.